File: Hospital.Server/Entities/Response/LabExamResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class LabExamResponse
    {
        public long Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal DefaultAmount { get; set; }
        public string ReferenceRange { get; set; } = string.Empty;
        public string Unit { get; set; } = string.Empty;
        public long? LaboratoryId { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
    }
}

File: hospital.client/src/types/LabExamResponse.ts
import type { LaboratoryResponse } from "./LaboratoryResponse";

export interface LabExamResponse {
  id: number;
  name: string;
  description: string;
  defaultAmount: number;
  referenceRange?: string | null;
  unit?: string | null;
  laboratoryId: number;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;

  laboratory?: LaboratoryResponse | null;
}

export interface LabExamRequest {
  id?: number | null;
  name?: string | null;
  description?: string | null;
  defaultAmount?: number | null;
  referenceRange?: string | null;
  unit?: string | null;
  laboratoryId?: number | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}
